src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+      removeAllButtonLabel,
+      visualPlaceholder,
+      emptyItemsLabel,
+      defaultSelectedItems,
+      onChange: propOnChange,
+      debounce,
+      status,
+      statusText,
+      selectedItems: controlledItems,
+      onItemSelect,
+      onRemoveAll,
+      ...passProps
+    } = this.props;
+
+    const selectedAndDisabledKeys = Object.keys(selectedKeys).reduce(
+      (amount: number, key: string) => {
+        if (key in disabledKeys) {
+          return amount + 1;
+        }
+        return amount;
+      },
+      0,
+    );
+
+    const showChipList =
+      chipListVisible && selectedItems && selectedItems.length > 0;
+
+    const showRemoveAllButton =
+      removeAllButtonLabel &&
+      Object.keys(selectedKeys).length > 0 &&
+      Object.keys(selectedKeys).length !== selectedAndDisabledKeys;
+
+    return (
+      <HtmlDiv
+        role="combobox"
+        aria-haspopup="listbox"
+        aria-controls={`${id}-popover`}
+        aria-expanded={showPopover}
+        {...passProps}
+        className={classnames(baseClassName, className, {
+          [comboboxClassNames.open]: showPopover,
+          [comboboxClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv className={classnames(comboboxClassNames.wrapper, {})}>
+          <Debounce waitFor={debounce}>
+            {(debouncer: Function) => (
+              <FilterInput
+                id={id}
+                labelText={labelText}
+                items={propItems}
+                onFilter={(filtered) =>
+                  this.setState({ filteredItems: filtered })
+                }
+                filterFunc={filter}
+                forwardedRef={this.filterInputRef}
+                onFocus={() => setPopoverVisibility(true)}
+                onKeyDown={handleKeyDown}
+                onBlur={handleBlur}
+                value={filterInputValue}
+                onChange={(value: string | undefined) => {
+                  if (propOnChange) {
+                    debouncer(propOnChange, value);
+                  }
+                  this.setState({ filterInputValue: value });
+                }}
+                visualPlaceholder={visualPlaceholder}
+                status={status}
+                statusText={statusText}
+              />
+            )}
+          </Debounce>
+          <Popover
+            sourceRef={this.filterInputRef.current}
+            matchWidth={true}
+            id={`${id}-popover`}
+            portalStyleProps={{ backgroundColor: 'white' }}
+            onKeyDown={handleKeyDown}
+          >
+            {showPopover && (
+              <ComboboxItemList
+                forwardRef={this.popoverListRef}
+                onBlur={handleBlur}
+                aria-activedescendant={
+                  currentSelection ? `${id}-${currentSelection}` : undefined
+                }
+              >
+                {filteredItems.length > 0 ? (
+                  filteredItems.map((item) => {
```

Comment:
Is the check necessary? mapping an empty array should return nothing?

Reply:
As you can see it is used in ternary operator; if no items are in the array the `ComboboxEmptyItem` is shown. 

Of course it could be written otherwise. e.g having the condition for the empty item straight away.

Reply:
I missed the conditional empty item there at the end. Not sure if it would be cleaner with the map without any condition and then the condition right next to the empty item description. There should be no performance impact, so both are fine for me.